File: improver/src/main/java/com/improver/model/in/ProRequestReview.java
package com.improver.model.in;

import lombok.Data;

import jakarta.validation.constraints.Size;
import java.util.List;

import static com.improver.util.ErrorMessages.REQUEST_REVIEW_MESSAGE_SIZE_ERROR_MESSAGE;
import static com.improver.util.ErrorMessages.REQUEST_REVIEW_SUBJECT_SIZE_ERROR_MESSAGE;
import static com.improver.util.database.DataRestrictions.*;

@Data
public class ProRequestReview {

    private List<String> emails;

    @Size(message = REQUEST_REVIEW_SUBJECT_SIZE_ERROR_MESSAGE, min = REVIEW_REQUEST_SUBJECT_MIN_SIZE, max = REVIEW_REQUEST_SUBJECT_MAX_SIZE)
    private String subject;

    @Size(message = REQUEST_REVIEW_MESSAGE_SIZE_ERROR_MESSAGE, min = REVIEW_REQUEST_MESSAGE_MIN_SIZE, max = REVIEW_REQUEST_MESSAGE_MAX_SIZE)
    private String message;
}

File: browser/src/app/api/models/ProRequestReview.ts
export class ProRequestReview {
  emails: Array<string>;
  subject: string;
  message: string;

  constructor(emails: string, subject: string, message: string) {
    this.emails = emails.split(',').map(email => email.trim());
    this.subject = subject;
    this.message = message;
  }
}
